worker/serve.go

```diff
@@ -0,0 +1,118 @@
+package worker
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/gabriel-vasile/mimetype"
+	"github.com/gotd/contrib/http_range"
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/api"
+)
+
+type (
+	// contentReader implements io.ReadSeeker but really Seek only returns the
+	// object's size, this reader can be used to pass to http.ServeContent but
+	// only if the caller made sure to manipulate the request in such a way that
+	// the only seeks are to find out the object's size
+	contentReader struct {
+		r    io.Reader
+		size int64
+	}
+)
+
+func newContentReader(r io.Reader, obj api.Object) (io.ReadSeeker, string, error) {
+	contentType := obj.ContentType()
+	if contentType == "" {
+		header := bytes.NewBuffer(nil)
+		mtype, err := mimetype.DetectReader(io.TeeReader(r, header))
```

Comment:
Do we still want to try and infer the type from the content or should we just call `mime.TypeByExtension` and call it a day?
Using the extension might be more effective for range requests since the name is always the same and ranges from the middle of a file might be missing the metadata required to infer the right type.

Reply:
hm yeah I guess that's a good idea 👍